finite_difference/src/field_mod.f90

```diff
@@ -291,6 +301,36 @@ end subroutine r2d_free_field
 
   !===================================================
 
+  function get_data(self) result(dptr)
+    use FortCL, only: read_buffer
+    !> Getter for the data associated with a field. Ensures that
+    !! the local copy is up-to-date with that on any remove
+    !! accelerator device (if using OpenACC or OpenCL).
+    class(r2d_field), target :: self
+    real(go_wp), dimension(:,:), pointer :: dptr
+    if(self%data_on_device)then
+       !$acc update host(self%data)
+       ! If FortCL is compiled without OpenCL enabled then this
+       ! call does nothing.
+       call read_buffer(self%device_ptr, self%data, &
+                        int(self%grid%nx*self%grid%ny, kind=8))
+    end if
+    dptr => self%data
+  end function get_data
+
+  !===================================================
+
+  function set_data(self, array) result(flag)
```

Comment:
Shouldn't this function write to device_ptr if using with the accelerator enabled?

Reply:
Yes, good point. However, FortCL has no `write_buffer()` method so I've created an Issue and a TODO for now.